src/dialogs/more-info/controls/more-info-weather.ts

```diff
@@ -50,12 +132,28 @@ class MoreInfoWeather extends LitElement {
     return false;
   }
 
+  protected updated(changedProps: PropertyValues): void {
+    super.updated(changedProps);
+    if (!this.stateObj || !this.hass) {
+      return;
+    }
+
+    if (this._needForecastSubscription() && !this._subscribed) {
+      this._subscribeForecastEvents();
```

Comment:
This ends up being called several times. I think the problem is that `this._subscribeForecastEvents()` is async, so when `updated` is called again `this._subscribed` is still not set.

awaiting `this._subscribeForecastEvents()` makes no difference, I guess there are then multiple calls to `updated` in parallel. 

What's the correct way to make sure we call `this._subscribeForecastEvents()` when needed?

`hui-weather-forecast-card` and `hui-weather-entity-row` have the same problem

Reply:
`_subscribed` should be a promise.

```ts
@state() private _subscribed?: Promise<() => void>;
```

Then you don't need to await when subscribing.

```ts
this._subscribed = subscribeForecast(
  this.hass!,
  stateObj.entity_id,
  forecastType,
  (event) => this._handleForecastEvent(event)
);
```

And you can unsubscribe like using the code below.

```ts
if (this._subscribed) {
  this._subscribed.then((unsub) => unsub?.());
  this._subscribed = undefined;
}
```

Similar pattern is used in `hui-history-graph-card`.